sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -25,7 +25,7 @@ public RecordMatcher(RecordedTestSanitizer sanitizer)
             "User-Agent"
         };
 
-        public virtual int FindMatch(Request request, IList<RecordEntry> entries)
+        public virtual int FindMatch(Request request, IList<RecordEntry> entries, out string failureMessage)
```

Comment:
I love the idea one thing I'm not sure about is adding a huge amount of allocations per match even in successful case.

Reply:
Keep it as is and I'll fix it up later.

Reply:
That's fair - we could duplicate this diffing logic and only build the error message if we failed to match.   I've just debugged enough of these that I'd like something.  😄